network/simulation/node_test.go

```diff
@@ -218,6 +219,69 @@ func TestAddNodes(t *testing.T) {
 	}
 }
 
+func TestAddBootNode(t *testing.T) {
+	sim := NewInProc(noopServiceFuncMap)
+	defer sim.Close()
+
+	id, err := sim.AddBootNode()
+	if err != nil {
+		t.Fatalf("Failed to add bootnode: %s", err)
+	}
+
+	bootNode := sim.Net.GetNode(id)
+	if !bootNode.Config.BootNode {
+		t.Fatalf("Bootnode did not have the respective flag in its config")
+	}
+
+	for _, node := range sim.Net.GetBootNodes() {
+		if !bytes.Equal(node.ID().Bytes(), bootNode.ID().Bytes()) {
+			t.Fatalf("Found an unexpected bootnode with ID: %s", node.ID().String())
+		}
+	}
+}
+
+func TestAddBootNodes(t *testing.T) {
+	bootNodeCount := 10
+
+	sim := NewInProc(noopServiceFuncMap)
+	defer sim.Close()
+
+	ids, err := sim.AddBootNodes(bootNodeCount)
+	if err != nil {
+		t.Fatalf("Failed to add bootnodes: %s", err)
+	}
+
+	bootNodes := sim.Net.GetBootNodes()
+	for _, id := range ids {
+		match := false
+		for _, node := range bootNodes {
+			if bytes.Equal(id.Bytes(), node.ID().Bytes()) {
+				match = true
+			}
+		}
+
+		if !match {
+			t.Fatalf("Added a bootnode with enode.ID, %s, but it was not found via GetBootNodes", id.String())
+		}
+	}
+}
+
+func TestAddNodesAndConnectToBootNode(t *testing.T) {
+	nodeCount := 20
+
+	sim := NewInProc(noopServiceFuncMap)
+	defer sim.Close()
+
+	ids, bootNodeID, err := sim.AddNodesAndConnectToBootNode(nodeCount)
+	if err != nil {
+		t.Fatalf("AddNodesAndConnectToBootNode Failed: %s", err)
+	}
+
+	// VerifyStar takes a map and an index to the bootnode, so append it and use the index
+	ids = append(ids, bootNodeID)
+	simulations.VerifyStar(t, sim.Net, ids, len(ids)-1)
```

Comment:
if you call this within `sim.AddNdesAndConnectToBootnode` you can omit this test

Reply:
`VerifyStar` is a tests helper function, so it can only be used in the test (and that's how the other `TestAddNodesAndConnect` functions do it, too).